pkgs/stdenv/generic/setup.sh

```diff
@@ -372,9 +372,11 @@ findInputs() {
     # so checking the array with "${!var}[@]" does not work
     # check if $pkgs is in the var ref array
     # TODO(@Ericson2314): Restore using associative array
-    if [[ "${var}[*]" = *" $pkg "* ]]; then
-        return 0
-    fi
+    for el in "${!var:-}[@]"; do
```

Comment:
Again, this has the same issue as the initial variant and will **not** expand the array.

Reply:
I did a quick test with
```
array=(one two three)
elementRef=two
ref=array
for el in "${!ref:-}[@]"; do
    echo "el: $el"
    if [[ "$el" = "$elementRef" ]]; then
        echo "success"
        return 0
    fi
done
```

I understand that I might be missing something here. If you have another test case that you feel fits better the situation, I'm interested. Otherwise feel free to ignore this PR if you think it's not useful.

Reply:
Oh wow, I just realised, I was testing in another shell, this doesn't work in bash. My apologies!

Reply:
Running that outputs:

```
el: one[@]
```

So exactly *not* what it's supposed to do, it *should* do something like this:

```bash
array=(one two three)
elementRef=two
ref=array
refSlice="$ref[*]"
for el in "${!refSlice-}"; do
  echo "el: $el"
  if [[ "$el" = "$elementRef" ]]; then
    echo "success"
    return 0
  fi
done
```

Output:

```
el: one two three
```

Reply:
> Oh wow, I just realised, I was testing in another shell, this doesn't work in bash. My apologies!
I was testing with osh, that I assumed to be equivalent, but in this case it is not.

No problem, at least we're now on the same page, that's at least a start.

Btw.: Thanks for doing the OSH stdenv work :-)

Reply:
I tested something new with bash, could you take a look when you have a moment?
it's using bash reference with `-n` introduced in bash 4.3
Since what we actually have is a reference, i thought we might as well make it explicit.
I tried to take into account the fact that the array might be undefined by going through an empty array.
Curious to have your thought.